ethsigner/commandline/src/main/java/tech/pegasys/ethsigner/EthSignerBaseCommand.java

```diff
@@ -168,13 +165,34 @@ public Duration getDownstreamHttpRequestTimeout() {
   }
 
   @Override
-  public Optional<PkcsStoreConfig> getClientCertificateOptions() {
-    return Optional.ofNullable(clientTlsCertificateOptions);
+  public Optional<DownstreamTlsOptions> getDownstreamTlsOptions() {
+    return Optional.ofNullable(picoCliTlsDownstreamOptions);
   }
 
-  @Override
-  public Optional<File> getWeb3ProviderKnownServersFile() {
-    return Optional.ofNullable(downstreamKnownServersFile);
+  /**
+   * Validate dependent options which are not enforceable by PicoCLI.
+   *
+   * <p>See https://github.com/remkop/picocli/issues/938
+   *
+   * <p>Note: This method will be removed once PicoCLI 4.2 is released.
+   *
+   * @param commandLine the CommandLine instance
+   * @param logger The Logger to use to print the warning
+   */
+  @Deprecated
```

Comment:
nit: can we add a message to the deprecated explaining when/why it should be removed?

Reply:
deprecate annotation doesn't take a message. Thats why I left a note in the javadoc of the method.